### PR TITLE
SE-3667: makes sure original usage_id is never a cloned pathway child

### PR DESCRIPTION
**Pull request by eLRuLL**
Testing instructions:
- Follow readme "Installation on Devstack" instructions
- Get blockstore test up: (`make testserver` inside the blockstore project)
- Follow readme "Testing on Devstack" instructions
- Callling the /api/lx-pathways/v1/pathway/{pathway_id} endpoint with the patch method and providing the a pathway child item ID as the one to clone, should return validation errors, as we should always allow original assets to be the ones pathway children are based on (this is basically what the backend test indicates as well)

Reviewers:

- [x] @swalladge 
- [ ] @symbolist

### CODE REVIEW

File: lx_pathway_plugin/tests/test_pathway_api.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def setUpClass(cls):
             description="",
             allow_public_learning=True,
             allow_public_read=True,
+            library_type='complex',
+            library_license='',
```
Comment (samuelallan72) [inline lx_pathway_plugin/tests/test_pathway_api.py:47]: Why was this this needed to be added?

File: lx_pathway_plugin/models.py [hunk 2]
```diff
@@ -123,6 +123,12 @@ def get_usage_id(self, obj):
             usage_id = obj["id"]
             return str(PathwayUsageLocator(pathway_key=pathway_key, block_type=block_type, usage_id=usage_id))
 
+    def validate_original_usage_id(self, obj):
+        locator = UsageKey.from_string(obj)
```
Comment (samuelallan72) [inline lx_pathway_plugin/models.py:130]: Please add a comment/docstring to explain this validator @eLRuLL

**Review by samuelallan72 — APPROVED** [review]
@eLRuLL :+1:

Just a couple of minor nits/questions, and then good to merge.

- [x] I tested this: tested adding a child item with an invalid original usage key and verified the change was rejected, tested normal operation with valid keys.
- [x] I read through the code
- [ ] NA I tested that the UI can be used with a keyboard only (tab order, keyboard controls).
- [x] Includes tests for bugfixes and/or features added.
- [x] Includes documentation

File: lx_pathway_plugin/models.py (hunk 2 above)
Comment (eLRuLL) [reply lx_pathway_plugin/models.py:130]: :+1:

File: lx_pathway_plugin/tests/test_pathway_api.py (hunk 1 above)
Comment (eLRuLL) [reply lx_pathway_plugin/tests/test_pathway_api.py:47]: it's necessary on the latest `create_library` method on edX

**Comment by symbolist** [comment]
Nice! 🙂

**Pull request merged**